### ISSUE TITLE
require-returns-check returning false errors

### ISSUE THREAD

**Issue by rpastro** [gajus/eslint-plugin-jsdoc#924]
## Expected behavior

There should be no `jsdoc/require-returns-check` errors reported for the functions below.  This was working correctly with eslint-plugin-jsdoc v39.3.14.

```js
/**
 * Return a V1 style query identifier.
 *
 * @param {string} id - The query identifier.
 * @returns {string} V1 style query identifier.
 */
function v1QueryId(id) {
    switch (id.substring(0, 4)) {
        case 'addq':
        case 'aliq':
        case 'locq':
            return id.substring(3);
        case 'lost':
            return id.substring(4);
        default:
            return id;
    }
}

/**
 * Parses the required header fields for the given SIP message.
 *
 * @param {string} logPrefix - The log prefix.
 * @param {string} sipMessage - The SIP message.
 * @param {string[]} headers - The header fields to be parsed.
 * @returns {object} Object with parsed header fields.
 */
function parseSipHeaders(logPrefix, sipMessage, headers) {
    try {
        return esappSip.parseHeaders(sipMessage, headers);
    } catch (err) {
        logger.error(logPrefix, `Failed to parser SIP header(s): ${headers}. ${err}\nMessage:${sipMessage}`);
        return {};
    }
}
```

## Actual behavior

After upgrading to v39.3.19 both functions above return a `jsdoc/require-returns-check` error.

```
/test/common.js
  125:1  error  JSDoc @returns declaration present but return expression not available in function  jsdoc/require-returns-check

/test/eventWriter.js
  46:1  error  JSDoc @returns declaration present but return expression not available in function  jsdoc/require-returns-check
```

## ESLint Config

```json
    "rules":   {
        "jsdoc/require-returns-check": 2
    }
```

## Environment

- Node version: 18.11.0
- ESLint version: 8.26.0
- `eslint-plugin-jsdoc` version: 39.3.19

**Comment by brettz9** [comment]
Thanks for the report. Some changes needed to be made in an earlier release as the rule was not catching missing return statements from certain branches of execution (if at least one valid return was present). The new logic may still have some gaps, so feel free to report any other such issues. Thanks!
